### ISSUE TITLE
GHCJS support

### ISSUE THREAD

**Issue by srid** [mmark-md/mmark#54]
Would you open for a PR to support GHCJS?

The `yaml` dependency doesn't compile on GHCJS.

```
test/Data/Yaml/IncludeSpec.hs:7:1: warning: [-Wdeprecations]
    Module ‘Data.Yaml’:
      GHCJS is not supported yet (will break at runtime once called).
  |
7 | import           Data.Yaml (ParseException(InvalidYaml))
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
[2 of 4] Compiling Data.Yaml.THSpec ( test/Data/Yaml/THSpec.hs, dist/build/spec/spec-tmp/Data/Yaml/THSpec.js_o )
Linking Template Haskell ()
Linking Template Haskell (Data.Yaml.IncludeSpec,ThRunner1)
uncaught exception in Haskell main thread: ReferenceError: h$yaml_parser_initialize is not defined
ReferenceError: h$yaml_parser_initialize is not defined
 
```

**Comment by mrkkrp** [comment]
How do you propose to solve this? Do we have a YAML library written in pure Haskell?

**Comment by srid** [comment]
I haven't taken a look at how exactly YAML is being used in mmark, but at first glance I'd say by using a different library that is written in pure Haskell. This comes up: https://github.com/haskell-hvr/HsYAML

**Comment by mrkkrp** [comment]
I’ll take a look.

**Comment by mrkkrp** [comment]
Hmm, pure YAML implementation is good, but that one lacks Aeson compatibility and parse errors are plain `String`s.

**Comment by srid** [comment]
Time for a `myaml` library then? :-)

**Comment by mrkkrp** [comment]
Heh, would be good to have something like `HsYAML` but with Megaparsec and Aeson-based ;)

**Comment by srid** [comment]
I don't remember if I already asked you this, but would it be reasonable to make `yaml` an optional dependency of `mmark` (so that it can be used in GHCJS today)?

**Comment by mrkkrp** [comment]
How will it work when `yaml` is disabled? Should it just ignore YAML blocks then?

**Comment by srid** [comment]
Yup that’s what I was thinking. Just ignore yaml blocks (or just extract
them as raw text).

**Comment by mrkkrp** [comment]
I think ignoring the blocks would be a better idea because that doesn't lead to a conditional API.

**Comment by benkolera** [comment]
@srid If you still care about this, I put in a PR #61 which I have working in a WIP obelisk implementation of the 'realworld'/conduit demo app in this commit: https://github.com/qfpl/reflex-realworld-example/commit/e8eb77943a8d20ce946a2fed44b4bffae7c5b975

**Comment by mrkkrp** [comment]
Implemented in #61.
